backend/dataall/modules/shares_base/db/share_object_repositories.py

```diff
@@ -40,14 +71,180 @@ def update_share_item_status_batch(
         share_uri: str,
         old_status: str,
         new_status: str,
+        share_item_type: ShareableType = None,
     ) -> bool:
-        (
+        query = session.query(ShareObjectItem).filter(
+            and_(ShareObjectItem.shareUri == share_uri, ShareObjectItem.status == old_status)
+        )
+        if share_item_type:
+            query = query.filter(ShareObjectItem.shareableType == share_item_type.value)
+
+        query.update(
+            {
+                ShareObjectItem.status: new_status,
+            }
+        )
+        return True
+
+    @staticmethod
+    def get_share_data(session, share_uri):
+        share: ShareObject = ShareObjectRepository.get_share_by_uri(session, share_uri)
+        dataset: DatasetBase = DatasetBaseRepository.get_dataset_by_uri(session, share.datasetUri)
+
+        source_environment: Environment = session.query(Environment).get(dataset.environmentUri)
+        if not source_environment:
+            raise exceptions.ObjectNotFound('SourceEnvironment', dataset.environmentUri)
+
+        target_environment: Environment = session.query(Environment).get(share.environmentUri)
+        if not target_environment:
+            raise exceptions.ObjectNotFound('TargetEnvironment', share.environmentUri)
+
+        env_group: EnvironmentGroup = (
+            session.query(EnvironmentGroup)
+            .filter(
+                and_(
+                    EnvironmentGroup.environmentUri == share.environmentUri,
+                    EnvironmentGroup.groupUri == share.groupUri,
+                )
+            )
+            .first()
+        )
+        if not env_group:
+            raise Exception(
+                f'Share object Team {share.groupUri} is not a member of the '
+                f'environment {target_environment.name}/{target_environment.AwsAccountId}'
+            )
+
+        source_env_group: EnvironmentGroup = (
+            session.query(EnvironmentGroup)
+            .filter(
+                and_(
+                    EnvironmentGroup.environmentUri == dataset.environmentUri,
+                    EnvironmentGroup.groupUri == dataset.SamlAdminGroupName,
+                )
+            )
+            .first()
+        )
+        if not source_env_group:
+            raise Exception(
+                f'Share object Team {dataset.SamlAdminGroupName} is not a member of the '
+                f'environment {dataset.environmentUri}'
+            )
+
+        return (
+            share,
+            dataset,
+            source_environment,
+            target_environment,
+            source_env_group,
+            env_group,
+        )
+
```

Comment:
**get_share_data_items_by_type** is new, it replaces get_share_data_items called in the data_sharing_service
```
(shared_tables, shared_folders, shared_buckets) = ShareObjectRepository.get_share_data_items(
                    session, share_uri, ShareItemStatus.Share_Approved.value
                )
```